plugin/storage/grpc/shared/plugin_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package shared
+
+import (
+	"context"
+	"testing"
+
+	"github.com/jaegertracing/jaeger/storage/dependencystore"
+	dependencyStoreMocks "github.com/jaegertracing/jaeger/storage/dependencystore/mocks"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	spanStoreMocks "github.com/jaegertracing/jaeger/storage/spanstore/mocks"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+)
+
+type mockStorageGRPCPlugin struct {
+	spanReader *spanStoreMocks.Reader
+	spanWriter *spanStoreMocks.Writer
+	depsReader *dependencyStoreMocks.Reader
+}
+
+type mockArchiveStoragePlugin struct {
+	archiveReader *spanStoreMocks.Reader
+	archiveWriter *spanStoreMocks.Writer
+}
+
+type mockStreamingSpanWriterPlugin struct {
+	streamWriter *spanStoreMocks.Writer
+}
```

Comment:
Since you never initialize the fields in these structs, you could've declared an empty dummy struct and define all functions on it returning nil

Reply:
hey @yurishkuro, didn't understand this...I got `File is not `goimports`-ed with -local github.com/jaegertracing/jaeger (goimports)
        "github.com/crossdock/crossdock-go/assert"`

after using assert and require both

Reply:
you need to use `stretchr/require`, not crossdock